### PR TITLE
For #8764: Update collection design on home

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

pass

collection2

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/TabInCollectionViewHolder.kt [hunk 1]
```diff
@@ -25,18 +28,19 @@ import mozilla.components.feature.tab.collections.Tab as ComponentTab
 class TabInCollectionViewHolder(
     val view: View,
     val interactor: CollectionInteractor,
+    private val differentLastItem: Boolean = false,
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/TabInCollectionViewHolder.kt]: Why do we need this new param if it seems like the only place we use it, we're setting it to true always?

Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/TabInCollectionViewHolder.kt]: This was a preparation for a try to use the same ViewHolder for other lists (that to not have a different UI for the last item), not just to have a common layout resource. I can remove it for the time being if it is confusing at this moment.

**Comment by mcarare** [comment]
> A few nits:
> 
> * From the mocks I think the collection collapsed icon is supposed to align with the favicons
> * The collapsed text and icon look a bit strange to me. Like not quite centered vertically and maybe the icon is a bit small? Might need some constraint edits
>   CC @topotropic to review UX beyond these nits

Correlating the design for favicon, list element and collection, there is no chance that the collection icon would be aligned with the favicons, with the specified dimensions:

- favicons have 40dp width and 16dp start padding 
         (https://user-images.githubusercontent.com/353960/75345843-cbc4bf00-589d-11ea-8197- 
         e9893dc061f7.png
         https://user-images.githubusercontent.com/353960/74059066-9e47cc80-49e7-11ea-8bb5- 
         705feea25388.png)
- collection icon has 24dp width and 16dp start padding 
         (https://user-images.githubusercontent.com/353960/75345953-fc0c5d80-589d-11ea-8f7f- 
         5f9605fad5d3.png)

**Comment by topotropic** [comment]
@mcarare You're right, that's a missing spec. 

The list item is in a card element, so it would be great if you could update left-margin from 16dp to 8dp for list items that are contained in a card. Ping me if you have questions. 
Thanks!

**Comment by mcarare** [comment]
Updated screenshot in comment 0 with new specs.

**Comment by liuche** [comment]
@ekager will re-review

**Pull request merged**